### PR TITLE
SDI-523: 🔒️ Fix security issues

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: audit-ci.json [hunk 1]
```diff
@@ -2,5 +2,5 @@
   "low": true,
   "package-manager": "auto",
   "registry": "https://registry.npmjs.org",
-  "allowlist": []
+  "allowlist": ["GHSA-9c47-m6qq-7p4h"]
```
Comment (petergphillips) [inline audit-ci.json:5]: ignored as will be fixed by https://github.com/github/advisory-database/pull/1541 anyway

File: integration_tests/tsconfig.json [hunk 2]
```diff
@@ -3,8 +3,10 @@
     "target": "es5",
     "noEmit": true,
     "lib": ["es5", "dom", "es2015.promise"],
-    "types": ["cypress"],
-    "esModuleInterop": true
+    "types": ["cypress", "express"],
```
Comment (petergphillips) [inline integration_tests/tsconfig.json:6]: needed so that express types are included

File: integration_tests/tsconfig.json [hunk 3]
```diff
@@ -3,8 +3,10 @@
...
+    "esModuleInterop": true,
+    "skipLibCheck": true,
```
Comment (petergphillips) [inline integration_tests/tsconfig.json:8]: this means that the redis errors in the node modules are ignored

File: integration_tests/tsconfig.json [hunk 4]
```diff
@@ -3,8 +3,10 @@
...
+    "typeRoots": ["../server/@types"]
```
Comment (petergphillips) [inline integration_tests/tsconfig.json:9]: needed as the Prison API mock api includes stuff from the server directory

File: server/@types/express/index.d.ts [hunk 5]
```diff
@@ -20,6 +20,11 @@ export declare global {
       verified?: boolean
       id: string
       logout(done: (err: unknown) => void): void
+      session: {
+        returnTo: string
+        nowInMinutes: number
```
Comment (petergphillips) [inline server/@types/express/index.d.ts:25]: not quite sure why the type check works for the `server` directory without this, but doesn't do any harm adding it anyway

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**